Graphics/ShaderTools/src/ShaderToolsCommon.cpp

```diff
@@ -327,4 +331,32 @@ bool ProcessShaderIncludes(const ShaderCreateInfo& ShaderCI, std::function<void(
     }
 }
 
+void MergeShaderIncludes(const ShaderCreateInfo& ShaderCI, IDataBlob** ppData)
+{
+    VERIFY_EXPR(ShaderCI.Desc.Name != nullptr);
+    VERIFY_EXPR(ppData != nullptr);
+
+    std::stringstream Stream;
+
+    auto IncludeHandler = [&](IDataBlob* pData, const Char* Path) {
+        Stream << String{static_cast<const Char*>(pData->GetConstDataPtr()), pData->GetSize()} << std::endl;
+    };
+
+    auto IncludeModify = [](Char* Start, Char* End) {
+        memset(Start, ' ', End - Start);
+    };
+
+    IncludePreprocessInfo Info{};
+    if (!ProcessShaderIncludes(ShaderCI, {IncludeHandler, IncludeModify}))
+    {
+        LOG_ERROR("Failed to merge includes for shader: '", ShaderCI.Desc.Name, "'.");
+        return;
+    }
+
+    const auto InlineData = Stream.str();
+    auto       pData      = DataBlobImpl::Create(InlineData.size(), InlineData.data());
```

Comment:
There is `StringDataBlobImpl` that can assume ownership of the `std::string` without making copies